### PR TITLE
Disallow unsupported covariate effect types in MFL parsing

### PR DESCRIPTION
**Pull request by JohanWBorg**
Raise error when parsing unknown covariate effects. As a placeholder for unsupported types, "CUSTOM" can be used to store an effect.

### CODE REVIEW

File: tests/tools/test_covsearch.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def test_validate_input_with_model(load_model_for_test, testdata, model_path):
         ),
         (
             ('nonmem', 'pheno.mod'),
-            dict(search_space='COVARIATE([CL, V], WGT, [EXP, ABC])'),
+            dict(search_space='COVARIATE([CL, V], WGT, [EXP, CUSTOM])'),
```
[the remark this answers is not in the document]
Comment (JohanWBorg) [reply tests/tools/test_covsearch.py:109]: No, the old test case would not give the same error. As the effect "ABC" would no longer be supported within MFL, the old test case would give a parsing error instead. This functionality is instead tested in `test_mfl.py` (see the changes made in that file)

**Comment by rikardn** [comment]
So `CUSTOM` means that there is an unknown effect?

**Comment by JohanWBorg** [comment]
@rikardn Yes, that would be the case. For instance, extracting the model information from the "pheno" example model would result in a custom effect, since we don't have a descriptor for the given effects.
```
model = load_example_model("pheno")
get_model_features(model)

Output
'ABSORPTION(INST);ELIMINATION(FO);COVARIATE([CL, V],WGT,CUSTOM,*);COVARIATE([V],APGR,CUSTOM,*)'
```
Another option would be to disallow the `CUSTOM` effect and not add them to the MFL object. If needed, they can still be extracted using `get_covariate_effects(model)`

**Comment by rikardn** [comment]
I think this sounds like a good way of handling custom effects for the time being.

**Pull request merged**